### PR TITLE
83 errfun should default to dt4 if not provided to tcplhit2 core

### PR DESCRIPTION
**Pull request by gracezhihuizhao**
'errfun' is required in `tcplhit2_core`. Adding a check for if this parameter is provided in `params` / output from `tcplfit2_core`. If the function should throw an error.

### CODE REVIEW

File: R/tcplhit2_core.R [hunk 1]
```diff
@@ -79,6 +79,8 @@ tcplhit2_core <- function(params, conc, resp, cutoff, onesd,bmr_scale = 1.349, b
 
   # get error distribution
   errfun = params[["errfun"]]
+  if (is.null(errfun))
+    warning("'errfun' is missing in the output from tcplfit2_core. 'errfun' tracks the error distribution assumed for the model fits and should be provided in 'params' list.")
```
Comment (sedavid01) [inline R/tcplhit2_core.R:83]: Suggest just adding "-- see tcplfit2_core help for additional details." at the end of the error function.

**Review by sedavid01 — COMMENTED** [review]
@gracezhihuizhao - Update looks good to me.  Just one minor suggestion provided.

@brown-jason:
* Once this update is done may want to have @cthunes test this update fixes the previous issue - unless it is deemed unnecessary from your end.
* Please notify if you think we should add a unit test to tcplfit2 for this edge case.

Thanks, 
-Sarah

**Review by sedavid01 — APPROVED** [review]
Approving changes.  @brown-jason please review and merge to 'dev' (via squash and merge) when you approve of the updates.

**Comment by cthunes** [comment]
It may be useful to either change `warning()` to `error()` to abort before running into another error, or to throw `warning()` and also set `errfun` to default `"dt4"` `errfun <- "dt4"`. What do guys you think?

**Comment by sedavid01** [comment]
> It may be useful to either change `warning()` to `error()` to abort before running into another error, or to throw `warning()` and also set `errfun` to default `"dt4"` `errfun <- "dt4"`. What do you guys think?

Ahhh, good catch.  Yes, this should throw an error rather than a warning.

File: R/tcplhit2_core.R [hunk 2]
```diff
@@ -79,6 +79,8 @@ tcplhit2_core <- function(params, conc, resp, cutoff, onesd,bmr_scale = 1.349, b
 
   # get error distribution
   errfun = params[["errfun"]]
+  if (is.null(errfun))
+    warning("'errfun' is missing in the output from tcplfit2_core. 'errfun' tracks the error distribution assumed for the model fits and should be provided in 'params' list. -- see tcplfit2_core help for additional details.")
```
Comment (sedavid01) [inline R/tcplhit2_core.R:83]: change to 'stop' rather than 'warning'

**Review by sedavid01 — CHANGES_REQUESTED** [review]

**Review by sedavid01 — APPROVED** [review]
Re-approving changes.

**Review by cthunes — APPROVED** [review]

**Comment by cthunes** [comment]
It threw an appropriate error when I didn't include the necessary parameter and when I did, I was able to process successfully. Thanks!
